cabal-testsuite/src/Test/Cabal/Run.hs

```diff
@@ -56,13 +56,24 @@ run _verbosity mb_cwd env_overrides path0 args = do
             return r
     withAsync drain $ \sync -> do
 
-    -- NB: do NOT extend this to take stdin; then we will
-    -- start deadlocking on AppVeyor.  See https://github.com/haskell/process/issues/76
-    pid <- runProcess path args mb_cwd mb_env Nothing {- no stdin -}
-                (Just writeh) (Just writeh)
+    let prc = (proc path args)
+          { cwd = mb_cwd
+          , env = mb_env
+          , std_in = case input of { Just _ -> CreatePipe; Nothing -> Inherit }
```

Comment:
Inheriting `stdin` is a bad idea?

Reply:
This mirrors the existing behaviour of [`runProcess`](https://hackage.haskell.org/package/process-1.6.10.0/docs/System-Process.html#v:runProcess)

> Handle to use for stdin (Nothing => use existing stdin)

Reply:
Happy enough to just create a pipe for it anyway regardless of whether or not its used

Reply:
Since this is mirroring existing behaviour this should be fine.